### PR TITLE
[CN-499] Add support for restoring from HotBackup resource name

### PR DESCRIPTION
**Pull request by mtyazici**
Changes:
- Agent upload status now additionally returns a `message` and `backupKey`. `backupKey` is the `tgz` file the agent uploaded to bucket.
-  Updated `uploadOptions`, replaced `BackupFolderPath` with `BackupBaseDir` and `MemberUUID` with `MemberID`.
- Removed `backupType` from `persistence` field because it became obsolete.
- Exposed `hotBackupResourceName` in `persistence` field and implemented the necessary changes in HotBackup and Hazelcast controllers.

Agent side counter-part PR: https://github.com/hazelcast/hazelcast-platform-operator-agent/pull/32

### CODE REVIEW

File: api/v1alpha1/hazelcast_types.go [hunk 1]
```diff
@@ -97,7 +99,7 @@ type HazelcastSpec struct {
 
 	// B&R Agent configurations
 	// +optional
-	// +kubebuilder:default:={repository: "docker.io/hazelcast/platform-operator-agent", version: "0.1.7"}
+	// +kubebuilder:default:={repository: "docker.io/mtypey/platform-operator-agent", version: "restore-hb-cr"}
```
Comment (mtyazici) [inline api/v1alpha1/hazelcast_types.go:102]: I will revert this when the agent side PR is merged.

File: test/e2e/hazelcast_persistence_test.go [hunk 2]
```diff
@@ -292,6 +301,55 @@ var _ = Describe("Hazelcast CR with Persistence feature enabled", Label("hz_pers
...
+		hazelcast.Spec.ExposeExternally = &hazelcastcomv1alpha1.ExposeExternallyConfiguration{
+			Type: hazelcastcomv1alpha1.ExposeExternallyTypeUnisocket,
+		}
+		CreateHazelcastCR(hazelcast)
+		evaluateReadyMembers(hzLookupKey, int(*hazelcast.Spec.ClusterSize))
+		WaitForMapSize(context.Background(), hzLookupKey, m.Name, 100, 1*Minute)
+
+	},
+		Entry("using GCP bucket", Label("slow"), "gs://operator-e2e-external-backup", "br-secret-gcp"),
+		Entry("using local backup", Label("slow"), Label("focus"), "", ""),
```
Comment (SeriyBg) [inline test/e2e/hazelcast_persistence_test.go:351]: `Label("focus")` also seems to be a leftover

**Review by SeriyBg — APPROVED** [review]
There are a few comments on the parts of the code that seems to be leftover just so that you don't forget about them. Other than that LGTM

**Review by hasancelik — APPROVED** [review]

File: api/v1alpha1/hazelcast_types.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (mtyazici) [reply api/v1alpha1/hazelcast_types.go:102]: I will revert it when agent side PR is merged

File: test/e2e/hazelcast_persistence_test.go (hunk 2 above)
Comment (mtyazici) [reply test/e2e/hazelcast_persistence_test.go:351]: That is right, I will delete it thanks.

**Pull request merged**